From ed39aa9e95a87570da084d2ea6696ca3d2b47d69 Mon Sep 17 00:00:00 2001
From: Ranjan Dutta <ranjan.dutta@intel.com>
Date: Tue, 7 Dec 2021 01:21:00 +0800
Subject: [PATCH 14/33] Revert "drm/nouveau/svm: Fix refcount leak bug and
 missing check against null bug"

This reverts commit 090e17075f6780dba35f91bf34f98170f29cb44f.
---
 drivers/gpu/drm/nouveau/nouveau_svm.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/drivers/gpu/drm/nouveau/nouveau_svm.c b/drivers/gpu/drm/nouveau/nouveau_svm.c
index f67700c028c7..1c3f890377d2 100644
--- a/drivers/gpu/drm/nouveau/nouveau_svm.c
+++ b/drivers/gpu/drm/nouveau/nouveau_svm.c
@@ -156,14 +156,10 @@ nouveau_svmm_bind(struct drm_device *dev, void *data,
 	 */
 
 	mm = get_task_mm(current);
-	if (!mm) {
-		return -EINVAL;
-	}
 	mmap_read_lock(mm);
 
 	if (!cli->svm.svmm) {
 		mmap_read_unlock(mm);
-		mmput(mm);
 		return -EINVAL;
 	}
 
-- 
2.27.0

